lib/src/compile.ts

```diff
@@ -89,8 +91,14 @@ function newCompileStringRequest(
 
   if (options?.url) input.setUrl(options.url.toString());
 
-  if (options && 'importer' in options && options.importer) {
-    input.setImporter(importers.register(options.importer));
+  const hasImporter = options && 'importer' in options && options.importer;
+  if (options?.url && !hasImporter) {
+    const importer = new proto.InboundMessage.CompileRequest.Importer();
+    importer.setPath(p.resolve('.'));
+    input.setImporter(importer);
```

Comment:
This makes sure the current directory in `FilesystemImporter('.')` is resolved on the host side instead of compiler side. It is not really necessary for the node host because as it would always restart the compiler so the compiler would have the same working directory as the host. However, as a reference implementation it is better to do the right thing here so that other implementations that re-use the compiler can follow the correct approach.